frontend/src/api/index.jsx

```diff
@@ -0,0 +1,57 @@
+const HOST = "http://localhost:5000/api/";
+
+const POST = async (endpoint, data, extraOptions) => {
+  const response = await fetch(HOST + endpoint, {
+    method: "POST",
+    cache: "no-cache",
+    credentials: "include",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify(data),
+    ...extraOptions,
+  });
+  return response;
+};
+
+const GET = async (endpoint, extraOptions) => {
+  const response = await fetch(HOST + endpoint, {
+    method: "GET",
+    cache: "no-cache",
+    credentials: "include",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    ...extraOptions,
+  });
+  return response;
+};
+
+const createError = (e, errMsg = null) => {
+  return { error: true, e, errMsg };
+};
+
+const createSuccess = (data) => {
+  return { error: false, data };
+};
+
+export const signUpUser = async (
+  firstName,
+  lastName,
+  email,
+  password,
+  isTutor
+) => {
+  const res = await POST("/auth/signUp", {
```

Comment:
will this double the forward slashes? (I.e. http://localhost:5000/api//auth/signUp). I'm not sure if that might be an issue

Reply:
Nevermind fixed! lgtm